app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,360 @@
+//! A module defining the special [`Entry`] type for the grid view of the breadcrumbs.
+
+use ensogl_core::display::shape::*;
+use ensogl_core::prelude::*;
+
+use ensogl_core::application::command::FrpNetworkProvider;
+use ensogl_core::application::frp::API;
+use ensogl_core::application::Application;
+use ensogl_core::data::color;
+use ensogl_core::display;
+use ensogl_core::display::scene::Layer;
+use ensogl_core::display::style::Data;
+use ensogl_core::Animation;
+use ensogl_grid_view::entry::Contour;
+use ensogl_grid_view::entry::EntryFrp;
+use ensogl_hardcoded_theme::application::component_browser::searcher::list_panel::breadcrumbs as theme;
+use ensogl_text as text;
+use ensogl_text::font::DEFAULT_FONT;
+
+
+// ==============
+// === Shapes ===
+// ==============
+
+/// A filled triangle pointing to the right. Used to separate breadcrumbs.
+pub mod separator {
+    use super::*;
+    use std::f32::consts::PI;
+
+    pub const ICON_WIDTH: f32 = 16.0;
+
+    ensogl_core::define_shape_system! {
+        above = [ensogl_grid_view::entry::shape];
+        pointer_events = false;
+        (style: Style, color: Vector4) {
+            let width = style.get_number(theme::separator::width);
+            let height = style.get_number(theme::separator::height);
+            let triangle = Triangle(width, height).rotate((PI/2.0).radians());
+            let shape = triangle.fill(color);
+            shape.into()
+        }
+    }
+}
+
+/// A three dots icon. Used to indicate that the last module in breadcrumbs list contains more
+/// ancestors.
+pub mod ellipsis {
+    use super::*;
+
+    pub const ICON_WIDTH: f32 = 32.0;
+
+    ensogl_core::define_shape_system! {
+        above = [ensogl_grid_view::entry::shape];
+        pointer_events = false;
+        (style: Style, alpha: f32) {
+            let radius = style.get_number(theme::ellipsis::circles_radius).px();
+            let gap = style.get_number(theme::ellipsis::circles_gap).px();
+            let background_width = style.get_number(theme::ellipsis::background_width);
+            let background_height = style.get_number(theme::ellipsis::background_height);
+            let background_corners_radius = style.get_number(theme::ellipsis::background_corners_radius);
+            let col = style.get_color(theme::ellipsis::circles_color);
+            let circles_color = Var::<color::Rgba>::rgba(col.red,col.green,col.blue,alpha.clone());
+            let bg_col = style.get_color(theme::ellipsis::background_color);
+            let background_color = Var::<color::Rgba>::rgba(bg_col.red,bg_col.green,bg_col.blue,alpha);
+
+            // `.repeat()` call would be more efficient than three separate circles, but it also
+            // requires `.intersection()` to crop the infinite grid of circles, which is
+            // currently broken. See https://www.pivotaltracker.com/story/show/182593513.
```

Comment:
Add "TODO" on the beginning, so it will be displayed in IntelliJ in eye-catching blue color.